fw/http_parser.c

```diff
@@ -973,6 +973,8 @@ process_trailer_hdr(TfwHttpMsg *hm, TfwStr *hdr, unsigned int id)
 	hdr->flags |= TFW_STR_TRAILER;
 	__set_bit(TFW_HTTP_B_CHUNKED_TRAILER, hm->flags);
 
+	hm->trailers_len += tfw_str_len(hdr) + 2;
```

Comment:
Seems `tfw_str_len` don't need here, you can simply do `hdr->len`, because `len` of "root" TfwStr always must contain length of all "child" chunks. Duplicates is not allowed here because you use currently parsing `hdr`.

And also, could you please replace magic number `2` here. As I understand it's length of CRLF, but IIRC in some cases we can have only one \n or \r, maybe `tfw_str_eolen` is more applicable here?

Reply:
> Seems tfw_str_len don't need here, you can simply do hdr->len, because len of "root" TfwStr always must contain length of all "child" chunks. Duplicates is not allowed here because you use currently parsing hdr.

okay, yes, I forgot this fact.

> could you please replace magic number 2 here. As I understand it's length of CRLF, but IIRC in some cases we can have only one \n or \r, maybe tfw_str_eolen is more applicable here?

No, for HTTP/1, the trailer chars are always `\r\n`, so the `len` is always `2`. I add some comment.

Reply:
> No, for HTTP/1, the trailer chars are always `\r\n`, so the `len` is always `2`. I add some comment.

You can use SLEN(S_CRLF) instead and it will be understandable without comments. 
And It's better to use a consistent comment style in one file.

Reply:
> No, for HTTP/1, the trailer chars are always `\r\n`, so the `len` is always `2`. I add some comment.

What is we actually receive `\n` only? It's prohibited by the RFC, but we saw clients and maybe servers, who actually send `\n` only instead of `\r\n`. Note also that a "server" could be just a some very home brew microservice.

Reply:
It fixed in a wrong way, `SLEN(S_CRLF)` always 2 bytes, but in some circumstances we can have only `\n`, this is allowed by our parser. You need to use `tfw_str_eolen()`. However you can't do it, in `process_trailer_hdr()`, because at this stage header is not fully parsed. I suggest to move this part to `tfw_http_msg_hdr_close()`.